packages/shorebird_cli/lib/src/git.dart

```diff
@@ -87,6 +87,29 @@ class Git {
     }
   }
 
+  /// List all branches in the git repository located at [directory] that match
+  /// the [pattern].
+  Future<String> listBranches({
+    required String directory,
+    required String pattern,
+  }) async {
+    final arguments = ['branch', '--all', '--list', pattern];
```

Comment:
This might behave unexpectedly in certain terminal environments due to colorization, etc. The equivalent plumbing command might be worth looking into https://stackoverflow.com/a/53504680/144755

Reply:
Good call, refactored to use `forEachRef` instead 👍